controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1052,6 +1055,16 @@ func (r *BareMetalHostReconciler) actionManageReady(prov provisioner.Provisioner
 	return r.manageHostPower(prov, info)
 }
 
+func getHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) (dirty bool, status *metal3v1alpha1.BareMetalHostStatus, err error) {
+	hostCopy := host.DeepCopy()
+	dirty, err = saveHostProvisioningSettings(hostCopy)
```

Comment:
Very likely the current method will not be invoked a lot of times, but it'd be nice to avoid copying the entire host when just the status would be sufficient

Reply:
`saveHostProvisioningSettings()` requires both the Spec and Status, so we'd have to pass them separately. The minuscule performance benefit doesn't seem worth the ugliness of that.